perception/tracking_object_merger/src/data_association/data_association.cpp

```diff
@@ -0,0 +1,382 @@
+    can_assign_matrix_ = can_assign_matrix_tmp.transpose();
+  }
+  {
+    const int max_dist_label_num = static_cast<int>(std::sqrt(max_dist_vector.size()));
+    Eigen::Map<Eigen::MatrixXd> max_dist_matrix_tmp(
+      max_dist_vector.data(), max_dist_label_num, max_dist_label_num);
+    max_dist_matrix_ = max_dist_matrix_tmp.transpose();
+  }
+  {
+    const int max_rad_label_num = static_cast<int>(std::sqrt(max_rad_vector.size()));
+    Eigen::Map<Eigen::MatrixXd> max_rad_matrix_tmp(
+      max_rad_vector.data(), max_rad_label_num, max_rad_label_num);
+    max_rad_matrix_ = max_rad_matrix_tmp.transpose();
+  }
+  {
+    const int min_iou_label_num = static_cast<int>(std::sqrt(min_iou_vector.size()));
+    Eigen::Map<Eigen::MatrixXd> min_iou_matrix_tmp(
+      min_iou_vector.data(), min_iou_label_num, min_iou_label_num);
+    min_iou_matrix_ = min_iou_matrix_tmp.transpose();
+  }
+  {
+    const int max_velocity_diff_label_num =
+      static_cast<int>(std::sqrt(max_velocity_diff_vector.size()));
+    Eigen::Map<Eigen::MatrixXd> max_velocity_diff_matrix_tmp(
+      max_velocity_diff_vector.data(), max_velocity_diff_label_num, max_velocity_diff_label_num);
+    max_velocity_diff_matrix_ = max_velocity_diff_matrix_tmp.transpose();
+  }
+
+  gnn_solver_ptr_ = std::make_unique<gnn_solver::MuSSP>();
+}
+
+void DataAssociation::assign(
+  const Eigen::MatrixXd & src, std::unordered_map<int, int> & direct_assignment,
+  std::unordered_map<int, int> & reverse_assignment)
+{
+  std::vector<std::vector<double>> score(src.rows());
+  for (int row = 0; row < src.rows(); ++row) {
+    score.at(row).resize(src.cols());
+    for (int col = 0; col < src.cols(); ++col) {
+      score.at(row).at(col) = src(row, col);
+    }
+  }
+  // Solve
+  gnn_solver_ptr_->maximizeLinearAssignment(score, &direct_assignment, &reverse_assignment);
+
+  for (auto itr = direct_assignment.begin(); itr != direct_assignment.end();) {
+    if (src(itr->first, itr->second) < score_threshold_) {
+      itr = direct_assignment.erase(itr);
+      continue;
+    } else {
+      ++itr;
+    }
+  }
+  for (auto itr = reverse_assignment.begin(); itr != reverse_assignment.end();) {
+    if (src(itr->second, itr->first) < score_threshold_) {
+      itr = reverse_assignment.erase(itr);
+      continue;
+    } else {
+      ++itr;
+    }
+  }
+}
+
+/**
+ * @brief calc score matrix between two tracked objects
+ *
+ * @param objects0 : measurements
+ * @param objects1 : base objects(tracker objects)
+ * @param debug_log
+ * @param file_name
+ * @return Eigen::MatrixXd
+ */
+Eigen::MatrixXd DataAssociation::calcScoreMatrix(
+  const autoware_auto_perception_msgs::msg::TrackedObjects & objects0,
+  const autoware_auto_perception_msgs::msg::TrackedObjects & objects1, const bool debug_log,
+  const std::string & file_name)
+{
+  // for debug
+  nlohmann::json log_data;
+  std::string log_file_name = "association_log.json";
+  if (!file_name.empty()) {
+    log_file_name = file_name;
+  }
```

Comment:
Too much debugging code in the code that is originally required is difficult to read.

Reply:
It might be better to have a debugger class.

Reply:
I removed debugging part, now it will be more comprehensive.